dataland-frontend/tests/component/components/pages/LandingPage.cy.ts

```diff
@@ -0,0 +1,112 @@
+import { checkButton, checkImage, checkAnchorByContent, checkAnchorByTarget } from "@ct/testUtils/ExistenceChecks";
+import NewLandingPage from "@/components/pages/NewLandingPage.vue";
+import { minimalKeycloakMock } from "@ct/testUtils/Keycloak";
+import content from "@/assets/content.json";
+import { type Page, type Section } from "@/types/ContentTypes";
+import { assertDefined } from "@/utils/TypeScriptUtils";
+
+describe("Component test for the landing page", () => {
+  it("Check if essential elements are present", () => {
+    cy.mountWithPlugins(NewLandingPage, {
+      keycloak: minimalKeycloakMock({
+        authenticated: false,
+      }),
+    }).then(() => {
+      validateTopBar();
+      validateIntroSection();
+      validateBrandsSection();
+
+      assertFrameworkPanelExists("Pathways to Paris");
+      assertFrameworkPanelExists("LkSG");
+      assertFrameworkPanelExists("EU Taxonomy");
+      assertFrameworkPanelExists("SFDR");
+      cy.get("button.joincampaign__button").should("exist");
+      cy.get("button.getintouch__text-button").should("exist");
+      checkNewFooter();
+    });
+  });
+});
+
+/**
+ * Validates the elements of the top bar
+ */
+function validateTopBar(): void {
+  checkImage("Dataland banner logo", getSingleImageNameInSection("Welcome to Dataland"));
+  checkButton("signup_dataland_button", "Sign Up");
+  checkAnchorByContent("Login");
+}
+
+/**
+ * Validates the elements of the intro section
+ */
+function validateIntroSection(): void {
+  checkImage(
+    "Liberate Data -  Empower Autonomy.   The alternative to data monopolies.",
+    getSingleImageNameInSection("Intro"),
+  );
+  cy.get("h1").should("contain.text", "Liberate Data");
+  cy.get("h1").should("contain.text", "Empower Autonomy");
+}
+
+/**
+ * Validates the images of the brands trusting in dataland
+ */
+function validateBrandsSection(): void {
+  const images = getLandingPageSection("Brands").image;
+  expect(images?.length).to.eq(6);
+  images!.forEach((image, index) => {
+    const filename = image.split("/").slice(-1)[0];
+    checkImage(`Brand ${index + 1}`, filename);
+  });
+}
+
+/**
+ * Check the new footer
+ */
+function checkNewFooter(): void {
+  /**
+   * Gets the footer element
+   * @returns the footer element
+   */
+  function getFooter(): Cypress.Chainable {
+    return cy.get("footer");
+  }
+  getFooter().should("exist");
```

Comment:
Seems like overkill.
Writing `cy.get("footer").should("exist")` is well readable and takes only one line.

Reply:
Absolutely, it was more useful in the past.